src/macros.rs

```diff
@@ -489,3 +496,67 @@ macro_rules! valkey_module {
         }
     }
 }
+
+/// Registers authentication callbacks with the Valkey module
+///
+/// Provides an unique safe wrapper for Valkey's authentication callback registration system.
+/// Used within the `valkey_module!` macro's auth field.
+///
+/// # Example
+/// ```ignore
+/// use valkey_module::valkey_module;
+///
+/// valkey_module! {
+///     name: "mymodule",
+///     version: 1,
+///     allocator: (ValkeyAlloc, ValkeyAlloc),
+///     auth: [
+///         blocking_callback,    // Called 2nd (LIFO)
+///         simple_callback,     // Called 1st
+///     ],
+///     // ... other fields
+/// }
+/// ```
+///
+/// # Note
+/// Callbacks are registered in LIFO order (last callback is called first).
+/// Requires Redis 7.2+ or Valkey 8.0+.
+#[macro_export]
+macro_rules! valkey_module_auth {
+    ($module_name:expr, $ctx:expr, $($auth_callback:expr),* $(,)*) => {
+        $(
+            {
+                paste::paste! {
+                    extern "C" fn [<__do_auth_ $module_name _ $auth_callback>](
+                        ctx: *mut $crate::raw::RedisModuleCtx,
+                        username: *mut $crate::raw::RedisModuleString,
+                        password: *mut $crate::raw::RedisModuleString,
+                        err: *mut *mut $crate::raw::RedisModuleString,
+                    ) -> std::os::raw::c_int {
+                        let context = $crate::Context::new(ctx);
+                        let ctx_ptr = unsafe { std::ptr::NonNull::new_unchecked(ctx) };
+                        let username = $crate::ValkeyString::new(Some(ctx_ptr), username);
+                        let password = $crate::ValkeyString::new(Some(ctx_ptr), password);
+
+                        match $auth_callback(&context, username, password) {
+                            Ok(result) => result,
+                            Err(e) => {
+                                let error_msg = $crate::ValkeyString::create_and_retain(&e.to_string());
+                                unsafe { *err = error_msg.inner };
+                                $crate::AUTH_HANDLED
+                            }
+                        }
+                    }
+
+                    #[cfg(all(any(
+                        feature = "min-redis-compatibility-version-7-2",
+                        feature = "min-valkey-compatibility-version-8-0"
+                    )))]
```

Comment:
Question: This will just skip the calling the API below if we are on versions before 7.2. Should we propagate an error back to the caller function instead to indicate that `valkey_module_auth` did not work instead of silently failing?